controllers/mirrorpeer_controller.go

```diff
@@ -106,6 +108,19 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	logger.V(2).Info("All validations for MirrorPeer passed", "MirrorPeer", req.NamespacedName)
 
+	if mirrorPeer.Labels == nil {
+		mirrorPeer.Labels = make(map[string]string)
+	}
+
+	if val, ok := mirrorPeer.Labels[hubRecoveryLabel]; !ok || val != "resource" {
+		logger.Info("Adding label to mirrorpeer")
+		mirrorPeer.Labels[hubRecoveryLabel] = "resource"
+		err = r.Client.Update(ctx, &mirrorPeer)
+		if err != nil {
+			logger.Info("Failed to update mirrorpeer", "error", err)
```

Comment:
```suggestion
			logger.Infof("Warning: Failed to update mirrorpeer %q. Error: %v", mirrorPeer.name, err)
```

Reply:
We need to differentiate different types of errors.
If it is NotFound error, that means resource is already deleted, so no need to continue with rest of reconcile.
If it is IsConflict error, that means resource is already updated, so we need to requeue and try again.
If it is other issue, log and continue.